rs/rosetta-api/icp/client/src/lib.rs

```diff
@@ -1250,3 +1315,49 @@ impl RosettaDisburseNeuronArgsBuilder {
         }
     }
 }
+
+pub struct RosettaNeuronInfoArgs {
+    pub neuron_index: u64,
+    pub public_key: Option<PublicKey>,
+    pub principal_id: Option<PrincipalId>,
+}
+
+impl RosettaNeuronInfoArgs {
+    pub fn builder(neuron_index: u64) -> RosettaNeuronInfoArgsBuilder {
+        RosettaNeuronInfoArgsBuilder::new(neuron_index)
+    }
+}
+
+pub struct RosettaNeuronInfoArgsBuilder {
+    neuron_index: u64,
+    public_key: Option<PublicKey>,
+    principal_id: Option<PrincipalId>,
+}
+
+impl RosettaNeuronInfoArgsBuilder {
```

Comment:
Just a curiosity, not a blocking comment.

I see a few places where we manually implement the Builder pattern. It seems like there are a few crates that provide automatic creation for them (like https://github.com/elastio/bon). Have we considered using them instead?

Reply:
We usually implement the building pattern ourselves in the entire repo. Generally speaking, in the monorepo new crates try to be avoided as it is another dependency to monitor and manage.